scripts/eval/evaluate_rag.py

```
@@ -0,0 +1,1002 @@
+        self.force_ingestion = force_ingestion
+        self.error_lock = Lock()  # Add a lock for thread-safe error counting
+        
+        print(f"    - Dataset path: {self.dataset_path}")
+        print(f"    - Evaluation data path: {self.eval_data}")
+
+    def check_collection_exists(self):
+        """Check if the collection exists in the vector database"""
+        url = f"{self.ingestor_server_url}/collections"
+        params = {"vdb_endpoint": self.vdb_endpoint}
+        try:
+            response = requests.get(url, params=params)
```

Ruff (0.15.10)

[error] Probable use of `requests` call without timeout

(S113)